pkg/app/AppListingService.go

```diff
@@ -770,6 +798,51 @@ func (impl AppListingServiceImpl) FetchAppStageStatus(appId int, appType int) ([
 	return appStageStatuses, err
 }
 
+func (impl AppListingServiceImpl) fetchCiArtifactAndGitTriggersMap(artifactIds []int) (CiArtifactAndGitCommitsMap map[int][]string, err error) {
+	artifacts, err := impl.ciArtifactRepository.GetByIds(artifactIds)
+	if err != nil {
+		return make(map[int][]string), err
+	}
+
+	CiArtifactAndGitCommitsMap = make(map[int][]string)
+	ciArtifactsWithMaterialsMap := make(map[int][]repository.CiMaterialInfo)
+	ciArtifactWithModificationMap := make(map[int][]repository.Modification)
+
+	for _, artifact := range artifacts {
+		//impl.ciArtifactRepository.GetCi
+		materialInfo, err := repository.GetCiMaterialInfo(artifact.MaterialInfo, artifact.DataSource)
```

Comment:
as artifacts are duplicate, this Unmarshal call will happen for same artifact more than once. This can be optimised by making artifact ids unique